cli.py

```diff
@@ -0,0 +1,282 @@
+
+def parseArgs():
+    parser = argparse.ArgumentParser()
+    # Folder path to annotations
+    parser.add_argument('--anno_gt', type=str)
+    parser.add_argument('--anno_det', type=str)
+
+    # Folder path to corresponding images:
+    parser.add_argument('--img', type=str, required=False)
+    #parser.add_argument('--img_gt', type=str, required=False)
+    #parser.add_argument('--img_det', type=str, required=False)
+
+    # format_gt: as in the application screenshot:
+    parser.add_argument('--format_gt', type=str)
+
+    # format_det: shares input from coord (either xyrb, xywh, or coco)
+    parser.add_argument('--format_det', type=str)
+
+    # Absolute or relative (abs, rel)
+    #parser.add_argument('--gtcoord',  type=str)
+    parser.add_argument('--coord_det', type=str)
+
+    # Actual computation type:
+    parser.add_argument('--metric', type=str)
+    
+    # metadata for metrics (not always needed)
+    parser.add_argument('--names', '-n', type=str, default='')
+    parser.add_argument('-t', '--threshold', type=float, default=0.5)
+    
+    # extra data (graphs and etc.)
+    parser.add_argument('--prgraph', '-pr', action='store_true')
+    parser.add_argument('-sp', '--savepath', type=str, required=False, default="./results/")
+    
+    # TODO: Add the same dataset analysis seen in the GUI:
+    #parser.add_argument('--info', action='store_true')
+    
+    return parser.parse_args()
+
+def verifyArgs(args):
+    if not os.path.exists(args.anno_gt):
+        raise Exception('--anno_gt path does not exist!')
+
+    if not os.path.exists(args.anno_det):
+        raise Exception('--anno_det path does not exist!')
+
+    if args.threshold > 1 or args.threshold < 0:
+        raise Exception('Incorrect range for threshold (0-1)')
+
+    if args.prgraph and args.savepath == '':
+        raise Exception("Precision-Recall graph specified but no save path given!")
+
+    if args.format_gt == 'voc' and args.names == '':
+        raise Exception("VOC or ImageNet ground truth format specified, but name file not specified.")
+
+    if 'tube' == args.format_gt != args.format_det:
+        raise Exception("Spatio-Temporal Tube AP specified in one format parameter but not other!")
+
+    
+    if args.img == '':
+        logging.warning("Image path not specified. Assuming path is same as ground truth annotations.")
+        args.img = args.anno_gt
+
+    if args.names == '':
+        logging.warning("Names property empty so assuming detection format is class_id based.")
+
+    if not os.path.exists(args.savepath):
+        logging.warning("Savepath directory %s is not found. Attempting to create folder"%(args.savepath))
+        try:
+            os.mkdir(args.savepath)
+        except:
+            logging.error("Could not create directory! Exiting...")
+            raise Exception()
+
+
+# TODO: Check if the plotting function is redundant for COCO eval
+def plot_coco_pr_graph(results, mAP=None, ap50=None, savePath=None, showGraphic=True):
```

Comment:
COCO's official tool does provide plots. Only PASCAL VOC does.
So, to avoid making users confused, we better only replicate what the official tools do.

The '--plot' argument should only generate the plot using the `pascal_voc_evaluator.plot_precision_recall_curve` function, if and only if `args.metric=coco`.

Reply:
I am assuming you mean that the VOC metric is the only one that can produce the plots? I removed `plot_coco_pr_graph` in 780b32cd8054d602ec7d9679ae0f4dcc9e4d8943